src/Monkeypatcher.cc

```diff
@@ -428,15 +434,36 @@ bool Monkeypatcher::try_patch_syscall(RecordTask* t) {
     return false;
   }
 
-  uint8_t following_bytes[256];
+  static const intptr_t MAXIMUM_LOOKBACK = 6;
+  uint8_t bytes[256 + MAXIMUM_LOOKBACK];
   size_t bytes_count = t->read_bytes_fallible(
-      ip.to_data_ptr<uint8_t>(), sizeof(following_bytes), following_bytes);
+      ip.to_data_ptr<uint8_t>() - MAXIMUM_LOOKBACK, sizeof(bytes), bytes);
+  if (bytes_count < MAXIMUM_LOOKBACK) {
+    LOG(debug) << "Declining to patch syscall at " << ip << " for lack of lookback";
+    tried_to_patch_syscall_addresses.insert(ip);
+    return false;
```

Comment:
Seems like doing lookback could make some syscalls that are currently patchable non-patchable. Should we do something about that?

Reply:
It could, if for some reason there aren't MAXIMUM_LOOKBACK bytes available before the syscall instruction. I'm not particularly concerned about this. If it becomes an issue we can be smarter.